MultiMarginCriterion.lua

```diff
@@ -9,10 +9,36 @@ end
 
 function MultiMarginCriterion:updateOutput(input, target)
    -- backward compatibility
+   local _target = target
+   if not torch.isTensor(_target) then
+     _target = input.new(1)
```

Comment:
I moved the 'tensorization' from the old C implementation to lua to have a single signature. I will add the classical lua 'or' conditional assignment trick.